<?xml version="1.0"?>
<launch>
    <!-- vehicle model and world -->
    <arg name="est" default="ekf2"/>
    <arg name="vehicle" default="iris"/>
    <arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/empty.world"/>
    
    <!-- UAV0 -->
    <group ns="uav0">
    
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="0"/>
        <arg name="fcu_url" default="udp://:14540@localhost:14580"/>
        
        <!-- Changing the GPS position (Will affect QGC location) -->
        <arg name="mavros/global_position/global/latitude" value="-27.415"/>
        <arg name="mavros/global_position/global/longitude" value="153.1"/>
        <arg name="mavros/global_position/global/altitude" value="0"/>
        
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn.launch">
        
            <arg name="x" value="-2"/>
            <arg name="y" value="1"/>
            <arg name="z" value="0"/>
            
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14560"/>
            <arg name="mavlink_tcp_port" value="4560"/>
            <arg name="ID" value="$(arg ID)"/>
            <arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))"/>
            <arg name="video_uri" value="$(eval 5600 + arg('ID'))"/>
            <arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))"/>
        </include>
        
        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>
        
        <node pkg="project" type="drone1_controller" name="drone1_controller_node" output="screen">
            <param name="fcu_url" value="$(arg fcu_url)"/>
            <param name="gcs_url" value=""/>
            
            <param name="init_x" value="-2"/>
            <param name="init_y" value="1"/>
            <param name="init_z" value="0"/>
            
            <param name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <param name="tgt_component" value="1"/>
        </node>
    </group>
    
</launch>

